rfd/XXXX-access-list-templates.md

```diff
@@ -0,0 +1,350 @@
+---
+author: Lisa Kim (lisa@goteleport.com)
+state: draft
+---
+
+# RFD XXXX - Templated Access Lists
+
+# Required Approvers
+
+- Engineering: @r0mant || @smallinsky (marek) && @kopiczko (pavel)
+- Product: @r0mant
+
+## What
+
+Access lists grant roles (and traits) to users on a long lived basis. Currently, to grant access through an access list, admins must select roles that already exists. This requires the admins to know what a role is and know how to create a role to customize access.
+
+This RFD proposes a new type of access list, a templated access list that removes the need for admins to know about roles. Templated access lists will take a simplified access specification and execute system behaviors such as Teleport creating the required roles and then Teleport assigning those roles to members and owners upon creating an Access List.
+
+## Why
+
+Improves Access List usability especially for day one users. Templated Access List allows an admin to focus on users and what resources users should have access to. It removes the need for an admin to learn how to create roles and removes how roles have relation to an Access List because Teleport will do it for them.
+
+### User story: As an admin, I want to create an access list that require members to request for short-term access to selected resources
+
+The template type to use for this case is `short-term`.
+
+Template type `short-term` represents an access list that utilizes JIT. Owners are reviewers. Members are requesters that are required to request access to resources and then upon approval are granted short-term access to requested Teleport resources.
+
+Admin will define what resources members will have access to by specifying the resource kinds and their labels and their resource principals.
+
+### User story: As an admin, I want to create an accesss list that grants long-term access to resources for members
+
+The template type to use for this case is `long-term`.
+
+Template type `long-term` represents an access list that grants members standing access to Teleport resources. Owners will have no special purpose other than to audit.
+
+Admin will define what resources members will have access to by specifying the resource kinds and their labels and their resource principals.
+
+This type of template is similar to how access list works now (non integrated types). Only difference is Teleport will create the necessary role for the admin.
+
+### Data Model
+
+#### Type Field
+
+In the current [AccessListSpec](https://github.com/gravitational/teleport/blob/bbb0f46b22ff88299908bef8dcf85d292aa379e1/api/proto/teleport/accesslist/v1/accesslist.proto#L75) model, there exists a field called `type`. We will introduce an additional type called `template` to indicate an access list used a template upon creation.
+
+```proto
+message AccessListSpec {
+  // ... other existing fields
+
+  // Existing type values: "" (default), "scim", and "static"
+  // NEW value: "template"
```

Comment:
I think "templated" would be a better name here. The list isn't technically a template, it's templated with the provided template_config.